src/rules/core/action.rs

```diff
@@ -0,0 +1,107 @@
+//! Actions of creatures.
+
+use serde::{Deserialize, Serialize};
+use std::fmt;
+
+/// An action is something that an actor is able to perform.
+/// Most often an action causes a change in the state of one or more objects.\
+/// Actions are identified by an id and can be of different nature.
+#[derive(Debug, Clone)]
+pub struct Action {
+    id: ActionId,
+    value: ActionValue,
+}
+
+impl Action {
+    fn new(id: ActionId, value: ActionValue) -> Self {
+        Self { id, value }
+    }
+
+    /// Returns a reference to `()` if this action is of the correct type, otherwise `None`.
+    pub fn movement(&self) -> Option<()> {
+        if let ActionValue::Movement = &self.value {
+            Some(())
+        } else {
+            None
+        }
+    }
+
+    /// Returns a reference to `()` if this action is of the correct type, otherwise `None`.
+    pub fn attack(&self) -> Option<()> {
```

Comment:
same

Reply:
done!

(I will do the same for statistics in another PR)